src/main/java/com/squareup/kotlinpoet/FunSpec.kt

```diff
@@ -133,6 +138,19 @@ class FunSpec private constructor(builder: Builder) {
 
   val isAccessor get() = name.isAccessor
 
+  private fun kdocWithParameters(): CodeBlock {
+    if (parameters.all { it.kdoc.isEmpty() }) {
+      return kdoc
+    }
+
+    return with (kdoc.toBuilder()) {
+      for (parameterSpec in parameters) {
+        add("@param %L %L", parameterSpec.name, parameterSpec.kdoc)
```

Comment:
Will this add `@param` lines even for parameters that don't have KDoc? I'd actually remove the `all` check at the top of the function body and do an `isEmpty` check for each parameter to skip those that don't have the KDoc.

Reply:
Fixed, thanks.